File: backend/app/domain/logic.py
from decimal import Decimal
from app.domain.constants import FEE_ROUNDING_MULTIPLE, ZERO_DECIMAL

def round_to_multiple_of_five(loan_amount: Decimal, fee: Decimal) -> Decimal:
    total = loan_amount + fee
    remainder = total % FEE_ROUNDING_MULTIPLE
    if remainder == ZERO_DECIMAL:
        return fee
    
    adjustment = FEE_ROUNDING_MULTIPLE - remainder
    return fee + adjustment

File: backend-go/internal/domain/logic.go
package domain

import (
	"github.com/shopspring/decimal"
)

func RoundToMultipleOfFive(loanAmount decimal.Decimal, fee decimal.Decimal) decimal.Decimal {
	total := loanAmount.Add(fee)
	remainder := total.Mod(FeeRoundingMultiple)
	if remainder.Equal(ZeroDecimal) {
		return fee
	}
	adjustment := FeeRoundingMultiple.Sub(remainder)
	return fee.Add(adjustment)
}
